app/controllers/state_file/questions/az_extension_payments_controller.rb

```diff
@@ -1,13 +1,17 @@
 module StateFile
   module Questions
     class AzExtensionPaymentsController < QuestionsController
-      def self.show?(intake)
-        Flipper.enabled?(:extension_period) && after_april_16_arizona_time?
+      # Class method called externally (e.g., in a flow)
+      def self.show?(intake, current_time: Time.current)
+        Flipper.enabled?(:extension_period) && after_filing_deadline?(intake, current_time)
       end
 
-      def self.after_april_16_arizona_time?
-        arizona_time = Time.use_zone("Arizona") { Time.zone.now }
-        arizona_time >= Time.use_zone("Arizona") { Time.zone.parse("2025-04-16 00:00:00") }
+      def self.after_filing_deadline?(intake, current_time)
+        StateInformationService.after_filing_deadline?(current_time, intake.state_code)
+      end
+
+      def show?
+        self.class.show?(current_intake, current_time: app_time)
```

Comment:
why do we need this one?

Reply:
bc app_time is an instance method we need this to be able to use it in self.show?

Reply:
this currently is unused